### ISSUE TITLE
Cross-Shaped Anti-Deadzone slider to fix analog snapping to axis.

### ISSUE THREAD

**Issue by RRDVD** [hrydgard/ppsspp#18691]
### What should happen

PPSSPP analog configuration should include a Cross-shaped Anti-deadzone Slider similar to the one found in Steam controller configutation that will allow PPSSPP users to manually "bridge the gap" from cardinal direction inputs on the analog, into their respective diagonals without the snapping as with current build. Currently, there is a large amount of dead space around the whole analog. The full range of motion in every diagonal, NE for instance, can be found between ~20 and ~70 degrees... as in there is a full 90 degrees of motion in only 50. An adjustable anti-deadzone to open up that small window to the whole diagonal range will Improve the smoothness of the experience and finer control in games.
here are screenshots of the Raw analog Input showing the current build:
![bad snapping without steam input](https://github.com/hrydgard/ppsspp/assets/99452449/db433d8d-ddd4-4e2d-98f8-22ef62bbbe45)
And now using Steam Input Implementation(not available for mobile):
![steam input working with Smooth in game diagonals-2](https://github.com/hrydgard/ppsspp/assets/99452449/4415e58c-e9c9-43a9-b19b-2efb2a45a40b)

In the first instance there is bad snapping to diags, in the second, the analog feels smooth and great.

### Who would this benefit

This will benefit all PPSSPP users who play games with an Analog Stick in games that support more than 8 directions. Specifically: Mobile PPSSPP users with gamepad, and PC Users who do not want to play through steam (using the steam input to correct the diagonals issue)

### Platform (if relevant)

Android

### Games this would be useful in

Phantasy Star Portable 2i, Kingdom Hearts, Final Fantasy, Ratchet and Clank, MGS, etc.

### Other emulators or software with a similar feature

Steam Controller Configuration has this implemented under cross-shaped deadzone with anti-deadzone.

### Checklist

- [X] Check the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already implemented.
- [X] [Search for other requests](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same feature.

**Comment by hrydgard** [comment]
The anti-deadzone ("low end radius") works considerably better in the current development builds (and [beta](https://www.ppsspp.org/docs/development/beta-testing/)) than in 1.16.6 . Can you try and see if that helps a bit?

I've never heard of cross shaped deadzones before, do you have some reference to how they work? PPSSPP currently has a square deadzone but thinking of adding circle too (we have that in tilt controls already).

**Comment by RRDVD** [comment]
I see. I will try out the dev build. 
It is essentially a t shape that extends the center dead zone to all four edges, able to widen or narrow as needed. but now that I look at it...  the cross shape will move out the inner corners considerably, perhaps instead of a cross, a pie wedge would be better. forgive me I am not familiar enough with functions to describe how it works.
![Cross Shaped deadzone](https://github.com/hrydgard/ppsspp/assets/99452449/e94cc768-59b7-43cd-b549-2257acb89499)

**Comment by hrydgard** [comment]
Yeah that would not be great for accurate directions. Pie wedges narrowing outwards, or simply a central round deadzone is probably the best.

**Comment by RRDVD** [comment]
I tested the low end radius with the cross deadzone it works even better than before.. feels like console accuracy! Yes, the central deadzone did not help with the gaps between forward and diagonal directions. A variable wedge shape would absolutely work perfectly.

(red zones excuded)
![wedge shaped deadzone](https://github.com/hrydgard/ppsspp/assets/99452449/7953800e-6557-40b5-86a1-c68993c31b9d)

**Comment by RRDVD** [comment]
Aha! I found a representation of what is happening.
https://youtu.be/OX0cP3DrwX4
In this video, the man describes Minecraft's left analog deadzone exactly how many games behave in ppsspp. It seems psp already uses a cross shaped deadzone?? Which makes sense when the steam implemented anti-deadzone fixes the axis-snapping and allows the player to perform smooth circles with the joystick in game. Both on touch controls and gamepads. I apologize if my earlier posts were confusing. I do not propose a new deadzone shape, only the ability to minimize a pre-existing one which is expressed by that cross-shaped zone.

**Comment by RRDVD** [comment]
![141191045-5b6520c5-980e-4df7-917c-d834a6c64a72~2](https://github.com/hrydgard/ppsspp/assets/99452449/9d893f18-80e4-4394-b65e-ed3ee742753d)
